### PR TITLE
feat: use big file downloader

### PR DESCRIPTION
**Pull request by alanshaw**
This PR uses `@storacha/unixfs-dl` to make ranged requests to big unixfs files in order to not exhaust the upstream worker of CPU time.

Also switches to using the `Link` type for CIDs instead of `string`. This allows us to detect if the CID is using the "raw" IPLD codec.

### CODE REVIEW

File: packages/ipfs-gateway-race/lib/request.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Determine if the request is for a specific byte range.
+ * @param {Headers} headers
+ */
+export const isRangeRequest = headers => headers.get('Range') != null
```
Comment (fforbeck) [inline packages/ipfs-gateway-race/lib/request.js:5]: what about using strict comparison here?

File: packages/ipfs-gateway-race/test/mocks/fixtures.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+exports.bigData = {
+  cid: 'bafybeihctj26zx2smrvl6ev7k2h4krjey3rd2irjsqtduhlowwe75fdzo4',
+  bytes: new TextEncoder().encode('Hello w3s.link!!!!!!'.repeat(10_000_000))
```
Comment (fforbeck) [inline packages/ipfs-gateway-race/test/mocks/fixtures.js:3]: Nice!

**Review by fforbeck — APPROVED** [review]
It looks good to me. I just added a minor comment about one of the comparisons with `null`.

File: packages/ipfs-gateway-race/lib/request.js (hunk 1 above)
Comment (alanshaw) [reply packages/ipfs-gateway-race/lib/request.js:5]:
```suggestion
export const isRangeRequest = headers => headers.get('Range') !== null
```

**Pull request merged**